src/aktualizr_lite/main.cc

```diff
@@ -256,6 +256,12 @@ static int daemon_main(LiteClient &client, const bpo::variables_map &variables_m
       data::ResultCode::Numeric rc = do_update(client, *target, lockfile);
       if (rc == data::ResultCode::Numeric::kOk) {
         current = *target;
+        client.http_client->updateHeader("x-ats-target", current.filename());
+        // Refreshing the metadata will cause us to report the x-ats-target
+        // up to the server so it can reflect our new status.
+        if (!client.primary->updateImagesMeta()) {
+          LOG_WARNING << "Unable to update device target information on server";
+        }
```

Comment:
What happens in case there is a network issue and the device is unable to report back to the server? Should it retry?

Reply:
The logic inside `updateImagesMeta` winds up calling (HttpClient::get)[https://github.com/advancedtelematic/aktualizr/blob/master/src/libaktualizr/http/httpclient.cc#L95] which has some retry logic. Its currently set to be (2)[https://github.com/advancedtelematic/aktualizr/blob/master/src/libaktualizr/http/httpclient.h#L66].

So worst case scenario is the device is online long enough to install update, but then drops its connection before it can check in. This means the device is running what it needs to be running, but its state won't be reflected in the UI until the next time it successfully checks in.

We could just put a `continue` statement after `client.http_client->updateHeader("x-ats-target", current.filename());`. Then it jumps to the start of our while loop and can use the code we already have for making this API call?

Reply:
This makes me wonder if we should sleep for less time in the event we fail to update the metadata. Right now it sleeps for the configured interval, maybe we should have it sleep for 30 seconds or something?

Reply:
Well, guess we can change this logic a bit to have sleep right at the start of while, and let the flow configure the amount of seconds to sleep, then we can start with sleep 0 (to avoid any sleep at the first run), and adjust depending on the case, such as failing to update the remote server. We can always sleep the default amount in case of a successful transmission, for example.

This would also reduce the need for multiple updateImagesMeta calls.

Reply:
okay - i've split this into 2 changes:

1) we now fix the header we report with and call "continue" to start the loop over
2) I changed the interval we wait when failing to check with the OTA server to be 10 seconds rather than the normal "successful interval".

I'd like to give this another test at home, but I think this might be a decent approach now.

Reply:
k - i've done some testing. It works. The benefit of a single place to check metadata made it easy to test the failure scenario - Just update /etc/hosts with a bad IP for ota-lite.foundries.io and run the daemon. You see it ping every 10s to get data rather than once an hour.